server/schema/article/article.resolver.js

```diff
@@ -197,26 +197,13 @@ module.exports = {
       );
       const _restritedPermission = UserPermission.exists(session, authToken, decodedToken, 'article.read.restricted');
 
-      // TODO: map the articles to return correct ones with errors instead of rejecting full output
-      if (
-        _articles.some(
-          (_article) =>
-            [ARTICLE_PUBLISH_TYPES.UNPUBLISHED, ARTICLE_PUBLISH_TYPES.ARCHIVED, ARTICLE_PUBLISH_TYPES.TRASHED].includes(
-              _article.publishStatus
-            ) && !_unpublishedPermission
-        )
-      ) {
-        throw APIError('NOT_FOUND', null, { reason: 'One or more of the requested articles cannot be found.' });
-      }
+      if (_restritedPermission && _unpublishedPermission) return _articles;
 
-      // TODO: map the articles to return correct ones with errors instead of rejecting full output
-      if (_articles.some((_article) => _article.isInstituteRestricted && !_restritedPermission)) {
-        throw APIError('FORBIDDEN', null, {
-          reason: 'One or more of the requested articles can only be viewed by students and faculty of NIT Rourkela.',
-        });
-      }
-
-      return _articles;
+      const publicArticles = _articles.filter(
+        ({ publishStatus, isInstituteRestricted }) =>
+          (_restritedPermission || !isInstituteRestricted) && (_unpublishedPermission || publishStatus)
+      );
+      return publicArticles;
```

Comment:
@120EE0692 please document the changes being made and the reasons behind them. Also, please add an error to the response so the frontend may know that not all articles were given.